Quiz/src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { Dispatch, SetStateAction, useContext } from "react";
+import "./Modal.css";
+import { Button } from "antd";
+import { Link } from "react-router-dom";
+import { QuizContext } from "../Helpers/Contexts";
+
+interface ModalProps {
+  setModalOpen: Dispatch<SetStateAction<boolean>>;
+}
+
+const Modal: React.FC<ModalProps> = ({ setModalOpen }) => {
+  const { setGameState } = useContext(QuizContext);
+  return (
+    <div className="modal">
+      <section className="modal-main">
+        <h2>Do you want to end the test?</h2>
+        <div className="modal-buttons">
+          <Button onClick={() => setModalOpen(false)}>Cancel</Button>
+          <Link to="/">
+            <Button
```

Comment:
if you use a button then do not use the link component since it can confuse other developers
please use router push inside button's handler